### PR TITLE
Dependency and dev-tooling upgrades (from OddBird).

### PR DESCRIPTION
**Pull request by jgerigmeyer**
Bugfixes:
- Fix horizontal scrollbar (https://github.com/oddbird/MetaDeploy/pull/1907)

Internal:
- Upgrade dependencies (https://github.com/oddbird/MetaDeploy/pull/1904)
- Use nvm and django-environ (https://github.com/oddbird/MetaDeploy/pull/1898)
- Switch from Dependabot to GH Actions (https://github.com/oddbird/MetaDeploy/pull/1900, https://github.com/oddbird/MetaDeploy/pull/1903)
- Switch from CircleCI to GH Actions (https://github.com/oddbird/MetaDeploy/pull/1899)
- Convert docs from .rst to .md (https://github.com/oddbird/MetaDeploy/pull/1910)

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
@davidjray These changes are very similar to https://github.com/SFDO-Tooling/Metecho/pull/2018, but will again require updates to branch protection (cc @davidmreed).

**Comment by davidmreed** [comment]
On it. We'll get this reviewed in the sprint starting tomorrow. Thanks, @jgerigmeyer!

**Comment by Br4nd0R** [comment]
@davidmreed - Happy to review this, in the upcoming sprint.

**Comment by davidmreed** [comment]
Thanks, @Br4nd0R - can you add a WI to capture that effort? Will be good to check in with @jstvz too to ensure that we get it integrated with the transition to self-hosted runners.

**Comment by Br4nd0R** [comment]
Yes, I'll go ahead put a WI in the next sprint for this.

File: .github/workflows/upgrade-deps.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: Upgrade dependencies
```
Comment (Br4nd0R) [inline .github/workflows/upgrade-deps.yml:1]: 👍

**Review by Br4nd0R — APPROVED** [review]
This looks great, thanks so much for the updates across the board as well as the new workflows!

File: pytest.ini [hunk 2]
```diff
@@ -1,17 +1,23 @@
 [pytest]
 python_files = *.py
 norecursedirs = .* _* node node_modules coverage venv
+junit_family = xunit2
```
[the remark this answers is not in the document]
Comment (jerivas) [reply pytest.ini]: Fixed in d3b309c5052b252ef8984d3de2caf09a893b52b6

**Pull request merged**